core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -792,6 +793,11 @@ private[spark] object Utils extends Logging {
   lazy val localIpAddress: String = findLocalIpAddress()
   lazy val localIpAddressHostname: String = getAddressHostName(localIpAddress)
 
+  def localIpURI(): String = {
+    val address = InetAddress.getLocalHost
```

Comment:
Maybe tie all IP logic to localIpAddress? `localIpAddressHostname` is already based on it.
Something like `InetAddress.getByName(localIpAddress)`